IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -87,7 +90,12 @@ - (BOOL)saveTokensWithConfiguration:(MSIDConfiguration *)configuration
     result = [self saveIDTokenWithConfiguration:configuration response:response context:context error:error];
 
     if (!result) return result;
-
+    
+    //Save App metadata
+    result = [self saveAppMetadataWithConfiguration:configuration response:response context:context error:error];
```

Comment:
if you save app metadata here, it won't be written when ADAL writes refresh tokens in MSAL format.